base/dict.jl

```diff
@@ -108,7 +108,7 @@ end
 Dict{K,V}(p::Pair) where {K,V} = setindex!(Dict{K,V}(), p.second, p.first)
 function Dict{K,V}(ps::Pair...) where V where K
     h = Dict{K,V}()
-    sizehint!(h, length(ps))
+    sizehint!(h, length(ps)::Int)
```

Comment:
This is surprising --- can we really not infer this?

Reply:
I can't recall where this came up. I might have just been routinely annotating all `sizehint!` calls. I can't find any evidence that we need this.